hiku/result.py

```diff
@@ -75,6 +75,18 @@ def _filter_fields(result, node):
     return {f.name: result[f.name] for f in node.fields}
 
 
+def _denormalize_record(record, result, node):
```

Comment:
I think that `_filter_fields` and `_denormalize_record` functions should be combined into recursive function with signature:
```
(type_, result, query_obj) -> Any
```
and which will be able to deal with all possible field types combinations

Reply:
rewritten in 55cc99a